tests/qnn/test_keras.py

```diff
@@ -34,24 +34,44 @@ def model(get_circuit, n_qubits, output_dim):
     layer1 = KerasLayer(c, w, output_dim)
     layer2 = KerasLayer(c, w, output_dim)
 
-    model = tf.keras.models.Sequential(
-        [
-            tf.keras.layers.Dense(n_qubits),
-            layer1,
-            tf.keras.layers.Dense(n_qubits),
-            layer2,
-            tf.keras.layers.Dense(output_dim),
-        ]
-    )
+    if isinstance(output_dim, tuple):
+
+        model = tf.keras.models.Sequential()
+        model.add(tf.keras.layers.Dense(n_qubits))
+        model.add(layer1)
+        model.add(tf.keras.layers.Flatten())
+        model.add(tf.keras.layers.Dense(n_qubits))
+        model.add(layer2)
+        model.add(tf.keras.layers.Flatten())
+        model.add(tf.keras.layers.Dense(output_dim[0] * output_dim[1]))
+
+    else:
+        model = tf.keras.models.Sequential(
+            [
+                tf.keras.layers.Dense(n_qubits),
+                layer1,
+                tf.keras.layers.Dense(n_qubits),
+                layer2,
+                tf.keras.layers.Dense(output_dim),
+            ]
+        )
```

Comment:
Also here, better to explicitly define a new fixture rather than add hidden logic